### PR TITLE
Copy over project and add readme

### PR DESCRIPTION
**Pull request by mariel-delagarza**
My final project!

### CODE REVIEW

File: fed-budget-chart/css/lines.css [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+p + p {
+  margin-top: 1rem;
+}
+
+#outerContainer {
+  display: grid;
+  grid-gap: .5rem;
+  grid-template-columns: 1fr;
+  align-items: start;
+  max-width: 1600px;
```
Comment (jnschrag) [inline fed-budget-chart/css/lines.css:23]: Let's set this to 1200px.

File: fed-budget-chart/css/lines.css [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+  margin: 0 auto;
+}
+
+.highcharts-background {
```
Comment (jnschrag) [inline fed-budget-chart/css/lines.css:27]: What's this for?

File: fed-budget-chart/css/lines.css [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+  width: fit-content;
+  min-width: 375px;
+  height: calc(100% - 0.4rem - 1px);
+  padding: 0.2rem 0.6rem;
+  font-family: "Source Sans Pro";
+  font-size: 14px;
+  background-color: #fff;
+  border: 2px solid #e7e7e7;
+  border-radius: 0;
+  /*box-shadow: inset 0 1px 3px 0 rgba(0, 0, 0, 0.05);*/
```
Comment (jnschrag) [inline fed-budget-chart/css/lines.css:84]: Go ahead and remove any commented out code.

File: fed-budget-chart/css/lines.css [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+  text-align: center;
+}
+
+button {
+  margin: 1 .5rem 1 .5rem;
+  padding: .3rem .5rem;
+  text-transform: uppercase;
+  font-family: 'Source Sans Pro';
+  border: 0;
+  transition: color 0.3s ease-in-out, background 0.3s ease-in-out;
```
Comment (jnschrag) [inline fed-budget-chart/css/lines.css:180]: Let's add `cursor: pointer` to all buttons.

File: fed-budget-chart/css/lines.css [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+}
+
+#submit {
```
Comment (jnschrag) [inline fed-budget-chart/css/lines.css:183]: Can you add a 2px border that is the same color as the background to the submit button? Right now it's a little bit smaller than the other two because it doesn't have a border.

File: fed-budget-chart/css/lines.css [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+  background-color: #243f3b;
+}
+
+.selectButtons {
+  background-color: transparent;
+  color: #000;
+  border: 2px solid #e7e7e7;
+}
+
+.selectButtons:hover {
```
Comment (jnschrag) [inline fed-budget-chart/css/lines.css:199]: For the hover effect on the select buttons, let's keep them transparent, but change the text & border color to the dark green.

File: fed-budget-chart/index.html [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    <script
+      type="text/javascript"
+      src="https://ajax.googleapis.com/ajax/libs/jquery/1.8.2/jquery.min.js"
+    ></script>
+    <script src="https://code.highcharts.com/highcharts.js"></script>
+    <script src="https://code.highcharts.com/highcharts-more.js"></script>
+    <script src="https://code.highcharts.com/6.0.7/modules/data.js"></script>
+    <script src="https://code.highcharts.com/maps/modules/exporting.js"></script>
+    <!-- Include iFrame Resizer -->
+    <!--<script src="https://csis-ilab.github.io/defense360-viz/globals/iframeSizer.contentWindow.min.js"></script>-->
```
Comment (jnschrag) [inline fed-budget-chart/index.html:51]: Please add the iFrame resizer back in.

File: fed-budget-chart/index.html [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    </main>
+
+    <!-- Include jQuery & Highcharts Libraries; Adjust per project -->
+    <script
+      type="text/javascript"
+      src="https://ajax.googleapis.com/ajax/libs/jquery/1.8.2/jquery.min.js"
+    ></script>
+    <script src="https://code.highcharts.com/highcharts.js"></script>
+    <script src="https://code.highcharts.com/highcharts-more.js"></script>
+    <script src="https://code.highcharts.com/6.0.7/modules/data.js"></script>
```
Comment (jnschrag) [inline fed-budget-chart/index.html:48]:
```suggestion
    <script src="https://code.highcharts.com/modules/data.js"></script>

```

File: fed-budget-chart/index.html [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+          <button id="submit">Submit</button>
+          <button class="selectButtons" id="select-all">Select All</button>
+          <button class="selectButtons" id="unselect-all">Unselect All</button>
+        </div>
+      </div>
+      <footer>CSIS Defense 360 | Source: OMB</footer>
+    </main>
+
+    <!-- Include jQuery & Highcharts Libraries; Adjust per project -->
+    <script
```
Comment (jnschrag) [inline fed-budget-chart/index.html:42]: Let's remove the jQuery dependency, it isn't needed.

File: fed-budget-chart/js/lines.js [hunk 10]
```diff
@@ -0,0 +1,265 @@
...
+function setUpCheckboxes(chart) {
+  const checkboxes = document.getElementById("checkboxes")
+  let series = chart.series
+  let checkboxesHTML = ""
+
+  for (let i = 0; i < series.length; i++) {
+    let isChecked = series[i].visible ? 'checked' : ""
+    checkboxesHTML += `
+    <div class="checkbox__wrapper" style="--color: ${series[i].color}">
+      <input type="checkbox" name="series" id="${i}" value="${i}" ${isChecked} style="--color: ${series[i].color}" />
```
Comment (jnschrag) [inline fed-budget-chart/js/lines.js:242]: You shouldn't need to add the inline style here. It will inherit from it's parent (the `.checkbox__wrapper`).

File: fed-budget-chart/css/lines.css (hunk 2 above)
Comment (mariel-delagarza) [reply fed-budget-chart/css/lines.css:27]: that's a good question! it's not in my current code so i must have taken it out after submitting the PR.

File: fed-budget-chart/js/lines.js (hunk 10 above)
Comment (mariel-delagarza) [reply fed-budget-chart/js/lines.js:242]: 🤔 I took it out of <input> but if I take it out of the <div> all of them turn black

**Comment by mariel-delagarza** [comment]
Hi hello - this evening I finally figured out how to solve the margin/x-axis issue. I fixed the width of the legend and the chart so both stay put and the legend no longer covers any data on the chart when resizing. The only remaining issue is a question for Seamus - the font-size for the x-axis is currently the default 11px, which is why FY2025 gets cut off at the end. It's completely visible at 10px but that seems pretty small. I also forced the dropdown to be "Discretionary Budget Authority by Agency" on load. It was previously displaying the data for Discretionary Budget Authority by Agency but the dropdown said "Discretionary Budget Authority by Function."

**Review by urchykli — APPROVED** [review]
